### ISSUE TITLE
[PM-1908] plan() from api masks submit errors

### ISSUE THREAD

**Issue by mayani** [pegasus-isi/pegasus#2021]
When using plan(submit=true) in the API, errors from HTCondor gets masked. With plan():

```
2023.03.14 19:07:44.751 CDT:   -----------------------------------------------------------------------
2023.03.14 19:07:45.537 CDT:   Database version: '5.0.4dev' (sqlite:////home/rynge/.pegasus/workflow.db)
2023.03.14 19:07:46.590 CDT:   Pegasus database was successfully created.
2023.03.14 19:07:46.595 CDT:   Database version: '5.0.4dev' (sqlite:////home/rynge/workflows/runs/rynge/pegasus/sequence-workflow/run0006/sequence-workflow-0.replicas.db)
2023.03.14 19:07:46.660 CDT:   Output replica catalog set to jdbc:sqlite:/home/rynge/workflows/runs/rynge/pegasus/sequence-workflow/run0006/sequence-workflow-0.replicas.db
[WARNING]  Submitting to condor sequence-workflow-0.dag.condor.sub
2023.03.14 19:07:46.997 CDT: [FATAL ERROR]
[1] java.lang.RuntimeException: Unable to submit the workflow using pegasus-run at edu.isi.pegasus.planner.client.CPlanner.executeCommand(CPlanner.java:659)
Traceback (most recent call last):
  File "./workflow.py", line 146, in <module>
    submit=True
  File "/usr/lib64/python3.6/site-packages/Pegasus/api/_utils.py", line 85, in wrapper
    assert f(self, *args, **kwargs) == None
  File "/usr/lib64/python3.6/site-packages/Pegasus/api/workflow.py", line 937, in wrapper
    return f(self, *args, **kwargs)
  File "/usr/lib64/python3.6/site-packages/Pegasus/api/workflow.py", line 1294, in plan
    **properties,
  File "/usr/lib64/python3.6/site-packages/Pegasus/client/_client.py", line 249, in plan
    rv = self._exec(cmd, stream_stdout=False, stream_stderr=True)
  File "/usr/lib64/python3.6/site-packages/Pegasus/client/_client.py", line 720, in _exec
    raise PegasusClientError("Pegasus command: {} FAILED".format(cmd), result)
Pegasus.client._client.PegasusClientError: Pegasus command: ['/usr/bin/pegasus-plan', '--cluster', 'horizontal', '--sites', 'condorpool', '--output-sites', 'local', '--staging-site', 'condorpool=stash', '--dir', '/home/rynge/workflows/runs', '--cleanup', 'inplace', '--submit', 'workflow.yml', '--json'] FAILED
```

Using pegasus-plan command line:

```
2023.03.14 19:08:31.137 CDT:   ----------------------------------------------------------------------- 
2023.03.14 19:08:34.782 CDT:   Database version: '5.0.4dev' (sqlite:////home/rynge/.pegasus/workflow.db) 
2023.03.14 19:08:35.770 CDT:   Pegasus database was successfully created. 
2023.03.14 19:08:35.776 CDT:   Database version: '5.0.4dev' (sqlite:////home/rynge/workflows/runs/rynge/pegasus/sequence-workflow/run0008/sequence-workflow-0.replicas.db) 
2023.03.14 19:08:35.827 CDT:   Output replica catalog set to jdbc:sqlite:/home/rynge/workflows/runs/rynge/pegasus/sequence-workflow/run0008/sequence-workflow-0.replicas.db 
2023.03.14 19:08:36.081 CDT:   Submitting to condor sequence-workflow-0.dag.condor.sub 
2023.03.14 19:08:37.429 CDT:   Error: Running ('/usr/bin/condor_submit', '-terse', 'sequence-workflow-0.dag.condor.sub') failed with 1 
2023.03.14 19:08:37.434 CDT:   Message: ERROR: Failed to commit job submission into the queue. 
2023.03.14 19:08:37.440 CDT:   ERROR: Submit requirement AssignAccountingGroup not met. 
2023.03.14 19:08:37.449 CDT: [FATAL ERROR]  
 [1] java.lang.RuntimeException: Unable to submit the workflow using pegasus-run at edu.isi.pegasus.planner.client.CPlanner.executeCommand(CPlanner.java:659) 
```

**Reporter:** @rynge
**Resolution:** Fixed
**Watchers:**
@rynge
@mayani
@vahi

**Comment by mayani** [comment]
**Author:** @vahi

@vahi should look at this and make sure what is happening.!

**Comment by mayani** [comment]
**Author:** @vahi

assigning it to rajiv.
rajiv please verify if this is fixed now. i think your fix went for 5.0.3. the version here is 5.0.4dev

**Comment by mayani** [comment]
**Author:** @mayani

@rynge Can you test and confirm it works for you?
